### PR TITLE
Chore: Add tests for ResultResource

### PR DESCRIPTION
**Pull request by maxr96**
### Checklist
- [x] Server: I added multiple integration tests (Spring) related to the features

### Motivation and Context

This PR increases the test coverage of the ResultResource class from 57%  to 91%. All REST endpoints are covered now.
Screenshot_2020-02-10 de tum in www1 artemis web rest(1)

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java [hunk 1]
```diff
@@ -294,23 +289,66 @@ public void createManualProgrammingExerciseResult() throws Exception {
...
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void createManualProgrammingExerciseResult_manualResultsNotAllowed() throws Exception {
+        var participation = setParticipationForProgrammingExercise(AssessmentType.AUTOMATIC);
+        Result result = ModelFactory.generateResult(true, 200).resultString("Good effort!");
+        List<Feedback> feedbacks = ModelFactory.generateFeedback().stream().peek(feedback -> feedback.setText("Good work here")).collect(Collectors.toList());
+        result.setFeedbacks(feedbacks);
+        result.setParticipation(participation);
+
+        String dummyHash = "9b3a9bd71a0d80e5bbc42204c319ed3d1d4f0d6d";
+        when(gitService.getLastCommitHash(ArgumentMatchers.any())).thenReturn(ObjectId.fromString(dummyHash));
```
Comment (krusche) [inline src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java]: wrong style, please use `doReturn...when`

Make sure to check other tests before implementing something wrong

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java [hunk 2]
```diff
@@ -294,23 +289,66 @@ public void createManualProgrammingExerciseResult() throws Exception {
...
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void createManualProgrammingExerciseResult_resultExists() throws Exception {
+        var participation = setParticipationForProgrammingExercise(AssessmentType.SEMI_AUTOMATIC);
+        Result result = ModelFactory.generateResult(true, 200).resultString("Good effort!");
+        List<Feedback> feedbacks = ModelFactory.generateFeedback().stream().peek(feedback -> feedback.setText("Good work here")).collect(Collectors.toList());
+        result.setFeedbacks(feedbacks);
+        result.setParticipation(participation);
+        result = resultRepository.save(result);
+
+        String dummyHash = "9b3a9bd71a0d80e5bbc42204c319ed3d1d4f0d6d";
```
Comment (krusche) [inline src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java]: if you mock this multiple times, please move it into the method
    @BeforeEach
    public void init() {}

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java [hunk 3]
```diff
@@ -294,23 +289,66 @@ public void createManualProgrammingExerciseResult() throws Exception {
...
+    }
+
+    @Test
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void createManualProgrammingExerciseResult_resultPropertyMissing() throws Exception {
+        var participation = setParticipationForProgrammingExercise(AssessmentType.SEMI_AUTOMATIC);
+        Result result = new Result();
+
+        String dummyHash = "9b3a9bd71a0d80e5bbc42204c319ed3d1d4f0d6d";
+        when(gitService.getLastCommitHash(ArgumentMatchers.any())).thenReturn(ObjectId.fromString(dummyHash));
```
Comment (krusche) [inline src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java]: wrong style

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java (hunk 1 above)
Comment (maxr96) [reply src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java]: Well I used the current test implementation. But yes I switched to this syntax. I see it offers a bit better type check.

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java (hunk 2 above)
Comment (maxr96) [reply src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java]: True. I moved the part of creating result and mocking with dummy hash into the `@BeforeEach` method

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java (hunk 3 above)
Comment (maxr96) [reply src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java]: Changed it

**Pull request merged**